File: leetcode3/변지협/v3/2525. Categorize Box According to Criteria.py

'''
1. 아이디어 :
문제에 나온 조건에 맞게 분류한다.

2. 시간복잡도 :
o(1)

3. 자료구조/알고리즘 :
'''

class Solution:
    def categorizeBox(self, length: int, width: int, height: int, mass: int) -> str:
        volume = length * width * height
        
        heavy = True if mass >= 100 else False
        bulky = False
        
        if length >= 10 ** 4 or width >= 10 ** 4 or height >= 10 ** 4 or volume >= 10 ** 9:
            bulky = True
        
        if heavy and bulky:
            return 'Both'
        elif not heavy and not bulky:
            return "Neither"
        elif bulky and not heavy:
            return "Bulky"
        else:
            return "Heavy"

File: leetcode3/황은지/2525. Categorize Box According to Criteria.js
/**
 * @param {number} length
 * @param {number} width
 * @param {number} height
 * @param {number} mass
 * @return {string}
 */
var categorizeBox = function (length, width, height, mass) {
  const bulky =
    length >= 10000 ||
    width >= 10000 ||
    height >= 10000 ||
    length * width * height >= Math.pow(10, 9);
  const heavy = mass >= 100;

  if (bulky) {
    if (heavy) return "Both";
    else return "Bulky";
  } else {
    if (heavy) return "Heavy";
    else return "Neither";
  }
};
